.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: tests
+
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  run_tests:
+    runs-on: macos-12
+    strategy:
+      matrix:
+        include:
+          # Latest xcode
+          - xcode: "13.4.1"
+            ios: "15.5"
+          # Older version
+          - xcode: "13.1"
```

Comment:
For now I set this to just an older version of Xcode, but ideally we should run the tests against the oldest version we want to support